resources/prosody-plugins/mod_persistent_lobby.lua

```diff
@@ -110,6 +110,10 @@ run_when_component_loaded(main_muc_component_host, function(host_module, host_na
                 return;
             end
 
+            if main_room and main_room._data.breakout_rooms_active then
```

Comment:
Instead of simply short circuiting the logic like this, it would make it a lot easier to read the flow of the code if we summarised the main room state as:

```lua
local main_room_active = main_room:has_occupant() or main_room._data.breakout_rooms_active;
```

Then replace the `not main_room:has_occupant()` below to `not main_room_active`.